include/numerics/petsc_vector.h

```diff
@@ -179,6 +179,8 @@ class PetscVector final : public NumericVector<T>
 
   virtual T sum () const override;
 
+  template <NormType N> Real lx_norm () const;
```

Comment:
No documentation, and if there was it would be "the argument here is a raw PETSc value"?

We need to document this and move it from `public:` to `private:` or `protected:`.

Reply:
So, this was something I did when I thought I needed l2 for matrices, and noticed the redundancy in the code, that's all. I agree it should be private, that's an oversight. Let me change that around and add some docs.